bundles/org.openhab.binding.powermax/src/main/java/org/openhab/binding/powermax/internal/connector/PowermaxConnector.java

```diff
@@ -106,14 +106,25 @@ public void handleIncomingMessage(byte[] incomingMessage) {
         }
     }
 
+    /**
+     * Handles a communication failure
+     */
+    public void handleCommunicationFailure() {
+        close();
+
+        for (int i = 0; i < listeners.size(); i++) {
+            listeners.get(i).onCommunicationFailure();
+        }
```

Comment:
You can possible simplify to:
```suggestion
        listeners.forEach(PowermaxMessageEventListener::onCommunicationFailure);
```

Reply:
Thanks, I have a general question about OH conventions for this. I agree that forEach is nicer but the new event handler is mostly a copy & paste of the existing one right above it. Where there's a newer coding convention available like this (more of a stylistic difference rather than functional)... what do you recommend for new code?

a) Duplicate the existing code to keep it consistent & improve readability
b) Use the new way even though it's inconsistent
c) Use the new way and update the existing code at the same time, even if it doesn't need to change

Different projects I've worked on each have their own approach to this, so if there's an OH recommendation, I'll try and follow it.

Reply:
I don't think we've written down this formally. But the general consensus is to use modern code constructions when possible. Because the code in this repository is already some years old you'll find plenty of older constructions. Specifically with openHAB 3 we switched to Java 11 which introduced more additional features. like `Set.of`. So in general there is already a lot of different coding styles.

So with regard to older code. I would say that B is preferred, so the code becomes better in general. And if you feel the need to change other code to do C. Like sometimes when working on code you see other constructs that look like bugs or don't completly use the openHAB framework correctly I tend to refactor the code.